src/test/ui/mismatched_types/issue-38371.stderr

```diff
@@ -0,0 +1,12 @@
+error[E0308]: mismatched types
+  --> $DIR/issue-38371.rs:14:8
+   |
+14 | fn foo(&foo: Foo) {  // illegal syntax
+   |        ^^^^ expected struct `Foo`, found reference
+   |
+   = note: expected type `Foo`
+   = note:    found type `&_`
```

Comment:
I don't think we have to fix this here necessarily, but I feel like these `note:` messages are kind of distracting from the `help`, which is more likely to be of use. But I'm not sure how to fix without doing potentially more harm than good, so let's leave as is.